src/transformation/visitors/delete.ts

```diff
@@ -5,21 +5,27 @@ import { transformLuaLibFunction, LuaLibFeature } from "../utils/lualib";
 import { unsupportedProperty } from "../utils/diagnostics";
 import { isArrayType, isNumberType } from "../utils/typescript";
 import { addToNumericExpression } from "../utils/lua-ast";
+import { transformOptionalDeleteExpression } from "./optional-chaining";
 
 export const transformDeleteExpression: FunctionVisitor<ts.DeleteExpression> = (node, context) => {
+    const innerExpression = ts.skipParentheses(node.expression);
```

Comment:
Oh, didn't know that.

Depending on how this works, though, this could result in incorrect behavior as parenthesis change the behavior of optional chaining, (a?.b)() is valid (if strict null checks is off). I'll investigate/test this.